src/ReactModuleContainerErrors.js

```diff
@@ -0,0 +1,53 @@
+export const ReactModuleContainerErrorNames = {
+  LazyComponentLoadingError: 'LazyComponentLoadingError',
+  FileAppenderLoadError: 'FileAppenderLoadError',
+  UnregisteredComponentUsedError: 'UnregisteredComponentUsedError',
+  UnregisteredMethodInvokedError: 'UnregisteredMethodInvokedError',
+  ListenerCallbackError: 'ListenerCallbackError'
+};
+
+export class ReactModuleContainerError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = this.constructor.name;
+
+    if (typeof Error.captureStackTrace === 'function') {
+      Error.captureStackTrace(this, this.constructor);
+    } else {
+      this.stack = (new Error(message)).stack;
+    }
+  }
+}
+
+export class UnregisteredMethodInvokedError extends ReactModuleContainerError {
+  constructor(methodName) {
+    super(`ModuleRegistry.invoke ${methodName} used but not yet registered`);
+  }
+}
+
+export class UnregisteredComponentUsedError extends ReactModuleContainerError {
+  constructor(componentId) {
+    super(`ModuleRegistry.component ${componentId} used but not yet registered`);
+  }
+}
+
+export class ListenerCallbackError extends ReactModuleContainerError {
+  constructor(methodName, error) {
+    super(`Error in listener callback of module registry method: ${methodName}`);
+    this.stack = this.stack + error.stack;
```

Comment:
This might be more complicated then this:
https://stackoverflow.com/questions/42754270/re-throwing-exception-in-nodejs-and-not-losing-stack-trace

Also maybe in addition or instead we should have `this.originalError = error`